runtime/build.rs

```diff
@@ -275,23 +275,23 @@ mod startup_snapshot {
   deno_core::extension!(
     runtime_main,
     deps = [runtime],
-    customizer = |ext: &mut deno_core::ExtensionBuilder| {
-      ext.esm(vec![ExtensionFileSource {
+    customizer = |ext: &mut deno_core::Extension| {
+      ext.esm_files.to_mut().push(ExtensionFileSource {
         specifier: "ext:runtime_main/js/99_main.js",
         code: deno_core::ExtensionFileSourceCode::IncludedInBinary(
           include_str!("js/99_main.js"),
         ),
-      }]);
-      ext.esm_entry_point("ext:runtime_main/js/99_main.js");
+      });
+      ext.esm_entry_point = Some("ext:runtime_main/js/99_main.js");
     }
   );
 
   #[cfg(feature = "snapshot_from_snapshot")]
   deno_core::extension!(
     runtime_main,
     deps = [runtime],
-    customizer = |ext: &mut deno_core::ExtensionBuilder| {
-      ext.esm_entry_point("ext:runtime/90_deno_ns.js");
+    customizer = |ext: &mut deno_core::Extension| {
```

Comment:
While we are in here, can we remove the customizer and update this to `esm_entry_point = `

Reply:
Oh huh. Just so I understand better, does the extension! macro not work w/`cfg`?

Reply:
Nope. But it might be as simple as changing the type to `expr`
```
    $(, esm_entry_point = $esm_entry_point:literal )?
    $(, esm_entry_point = $esm_entry_point:expr )?
```
```rust
  esm_entry_point = {
    #[cfg(feature = "snapshot_from_snapshot")]
    "ext:runtime/90_deno_ns.js"
    #[cfg(not(feature = "snapshot_from_snapshot"))]
    "ext:runtime_main/js/99_main.js"
  },
```
Lots of built-in macros like https://doc.rust-lang.org/std/macro.concat.html do that anyway so they get better type errors if the user passes in a non-literal.

Reply:
Oh I see what you mean -- but in the meantime we could use two copies of this extension and use 
```
extension!(...
esm_entry_point = ...)
```

Instead of the customer?

Reply:
Probably not, just because `extension!()` expands to
```rust
pub struct ext {}
impl ext {}
```
and the directive would only apply to the first one. I don't think we can wrap it in anything that would work without having to tamper with a bunch of stuff.

Reply:
Oh duh we can just remove the customizer in this case and let it be overrided in the other

Reply:
Yeah, your point about allowing #cfg inside extension is really good, however, and I definitely think that's worth looking at. It would simplify at least a few of these extensions.

Reply:
Okay nevermind I was looking at the wrong extension, there are already two copies of this one 🤦‍♂️